File: Leetcode/sbgo/Leetcode541-反转字符串.go
package sbgo

func reverseStr(s string, k int) string {
	bs, n := []byte(s), len(s)
	rev := func(i, j int) {
		for i < j {
			bs[i], bs[j] = bs[j], bs[i]
			i++
			j--
		}
	}
	for i := 0; i < n; i += (2 * k) {
		if i+k <= n {
			rev(i, i+k-1)
		} else {
			rev(i, n-1)
		}
	}
	return string(bs)
}

File: Leetcode/PY/Leetcode541-反转字符串.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        n,it= len(s),0
        s = list(s)
        def rev(l,r):
            while l < r:
                s[l],s[r] = s[r],s[l]
                l += 1
                r -= 1
        while it < n:
            ed = it + k - 1
            if ed < n:
                rev(it,ed)
            else:
                rev(it,n - 1)
            it = it + 2 * k
        return ''.join(s)